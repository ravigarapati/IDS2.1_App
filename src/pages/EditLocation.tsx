import React, {useEffect, useState} from 'react';
import {View, StyleSheet, Platform, Image, Keyboard} from 'react-native';
import {
  Button,
  CustomAutoCompleteInput,
  CustomInputText,
  CustomText,
} from '../components';
import {Enum} from '../utils/enum';
import {connect} from 'react-redux';
import {
  updateLocation,
  getLocationSuggestions,
} from '../store/actions/HomeOwnerActions';
import {Dictionary} from '../utils/dictionary';
import {FlatList, TouchableHighlight} from 'react-native-gesture-handler';
import {Typography} from '../styles';

const EditLocation = ({
  navigation,
  currentLocation,
  updateLocation,
  getLocationSuggestions,
  locationSuggestions,
  weatherInfoLocation,
}) => {
  const [location, setLocation] = useState(currentLocation);
  const [isChanged, setIsChanged] = useState(true);
  const [location2, setLocation2] = useState(weatherInfoLocation.city);
  const [showSuggestions, setShowSuggestions] = useState(false);
  const [selectedName, setSelectedName] = useState('');
  const [mData, setMData] = useState([]);
  const [selectedSuggestion, setSelectedSuggestion] = useState({});

  const save = () => {
    updateLocation(selectedSuggestion);
    navigation.navigate('HomeTabs');
  };

  const cancel = () => {
    navigation.navigate('HomeTabs');
  };

  const handleSelection = item => {
    setShowSuggestions(false);
    Keyboard.dismiss();
    setIsChanged(false);
    setSelectedSuggestion(item);
  };

  return (
    <View style={{flex: 1}}>
      <Image
        style={{height: 8, width: '100%'}}
        source={require('../assets/images/header_ribbon.png')}
      />
      <View style={styles.container}>
        <View style={{flexDirection: 'row'}}>
          <View style={{width: '100%'}}>
            <CustomInputText
              disableCache={false}
              testID="location"
              onChange={e => {
                getLocationSuggestions({location: e, countryCode: 'abc'});
                setLocation2(e);
                setShowSuggestions(true);
              }}
              value={location2}
            />
          </View>

          <Image
            //tintColor={dropDownIconColor}
            style={{
              marginLeft: '-6%',
              alignSelf: 'center',
            }}
            source={require('./../assets/images/search.png')}
          />
        </View>

        {showSuggestions && locationSuggestions.length != 0 && (
          <View
            style={[
              styles.overlay,
              {
                backgroundColor: 'white',
                shadowColor: '#000',
                shadowOffset: {
                  width: 0,
                  height: 2,
                },
                shadowOpacity: 0.25,
                shadowRadius: 3.84,
                marginTop: -10,
              },
            ]}>
            <View>
              <FlatList
                keyExtractor={(item, index) => index.toString()}
                keyboardShouldPersistTaps="always"
                initialNumToRender={4}
                onEndReachedThreshold={0.5}
                data={locationSuggestions}
                renderItem={({item}) => (
                  <TouchableHighlight
                    testID="suggestion"
                    onPressIn={() => {
                      setLocation2(item.Label);
                      setSelectedName(item.Label);
                    }}
                    onPressOut={() => {
                      setSelectedName('');
                      setSelectedSuggestion(item);
                      setIsChanged(false);
                      setShowSuggestions(false);
                    }}
                    onPress={() => {
                      handleSelection(item);
                    }}>
                    <View style={[styles.suggestionElementView]}>
                      <CustomText
                        style={[
                          styles.flatListElements,
                          selectedName !== '' && selectedName === item.Label
                            ? {
                                backgroundColor: 'rgba(0, 73, 117, 1)',
                                color: 'white',
                              }
                            : {},
                        ]}
                        text={item.Label}
                        align="left"
                      />
                    </View>
                  </TouchableHighlight>
                )}
              />
            </View>
          </View>
        )}

        <View>
          <Button
            testID="submit"
            disabled={isChanged}
            text={Dictionary.button.save}
            type={'primary'}
            onPress={save}
          />
          <Button
            testID="cancel"
            text={Dictionary.button.cancel}
            type={'secondary'}
            onPress={cancel}
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    paddingHorizontal: 15,
    paddingTop: 20,
    justifyContent: 'space-between',
  },

  textField: {
    width: '100%',
    paddingRight: 12,
    paddingLeft: 12,
    borderBottomWidth: Platform.OS === 'ios' ? 1.5 : 0,
    marginTop: Platform.OS === 'ios' ? '3%' : 0,
    fontSize: 16,
    paddingBottom: 14,
    fontFamily: Typography.FONT_FAMILY_REGULAR,
  },
  flatListElements: {
    paddingLeft: 7,
    paddingRight: 7,
    paddingTop: 14,
    paddingBottom: 14,
  },
  suggestionElementView: {
    width: '100%',
    paddingLeft: 8,
  },
  overlay: {
    width: '100%',
    height: '70%',
  },
});

const mapStateToProps = state => {
  return {
    currentLocation: state.homeOwner.location,
    locationSuggestions: state.homeOwner.locationSuggestions,
    weatherInfoLocation: state.homeOwner.weatherInfoLocation,
  };
};

const mapDispatchToProps = {
  updateLocation,
  getLocationSuggestions,
};

export default connect(mapStateToProps, mapDispatchToProps)(EditLocation);
