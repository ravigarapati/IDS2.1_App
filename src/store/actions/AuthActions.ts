import AsyncStorage from '@react-native-async-storage/async-storage';
import {
  CURRENT_USER,
  CURRENT_STEP,
  UPDATE_CURRENT_USER,
  LOGOUT,
  TERMS_AND_CONDITIONS,
  LATEST_VERSION,
  SET_STEP_TITLES,
  SET_USER_ROLE,
  SET_SHOW_ROLE,
  SET_USER_ADDRESS,
  RESET_USER,
  SET_USER_NAME,
} from '../labels/AuthLabels';
import {Auth} from 'aws-amplify';
import {showToast} from '../../components/CustomToast';
import {
  apiInterceptorForPost,
  apiInterceptorForPostCustomApi,
  openApiInterceptorForGet,
  apiInterceptorForPostAPI,
} from './CommonActions';
import ClearCache from 'oa-react-native-clear-cache';

export const setCurrentUser = user => {
  return async dispatch => {
    dispatch({type: CURRENT_USER, user: user});
    ClearCache.clearAppCache(() => {});
  };
};
export const setCurrentStep = stepNumber => {
  return async dispatch => {
    dispatch({type: CURRENT_STEP, data: stepNumber});
  };
};
export const updateUserAttributes = (userFromStore, data, callback?) => {
  return async dispatch => {
    Auth.updateUserAttributes(userFromStore, data)
      .then(() => {
        if (callback) {
          callback();
        }
      })
      .catch(error => {
        showToast(error.message, 'error');
      });
  };
};

export const verifyCurrentUserAttribute = attribute => {
  Auth.verifyCurrentUserAttribute(attribute)
    .then(() => {})
    .catch(error => {
      showToast(error.message, 'error');
    });
};

export const verifyCurrentUserAttributeSubmit = (
  attribute,
  value,
  callback?,
) => {
  Auth.verifyCurrentUserAttributeSubmit(attribute, value)
    .then(() => {
      if (callback) {
        callback();
      }
    })
    .catch(error => {
      showToast(error.message, 'error');
    });
};

export const updateUserObject = attributes => {
  return async dispatch => {
    dispatch({type: UPDATE_CURRENT_USER, data: attributes});
  };
};

export const checkAdminDetails = (attr, callback?) => {
  return async dispatch => {
    const path = '/checkadmindetails';
    apiInterceptorForPost(
      dispatch,
      path,
      attr,
      response => {
        callback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const createProfile = (userData, callback?) => {
  return async dispatch => {
    const path = '/registration';
    apiInterceptorForPost(
      dispatch,
      path,
      userData,
      response => {
        callback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const createProfileHomeOwner = (userData, callback?) => {
  return async dispatch => {
    const baseUrl =
      'https://r9nzt6y3ui.execute-api.us-east-1.amazonaws.com/development';
    apiInterceptorForPostCustomApi(
      dispatch,
      baseUrl,
      '/user/register',
      {
        firstName: userData.firstName,
        lastName: userData.lastName,
        phoneNumber: userData.phoneNumber,
        type: 'HomeOwner',
        timestamp: `${new Date().valueOf()}`,
      },
      response => {
        if (response.message) {
          if (
            response.message === 'Operation succeed' ||
            response.message.startsWith("'e")
          ) {
            callback();
          } else if (
            response.message === 'The user account has already registered'
          ) {
            showToast(response.message, 'info');
          } else {
            showToast(response.message, 'error');
          }
        }
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const logout = () => {
  return async dispatch => {
    dispatch({type: LOGOUT, data: {}});
    // clear the storage
    ClearCache.getAppCacheSize(data => {});
    ClearCache.clearAppCache(() => {});
    await AsyncStorage.removeItem('persist:root').then(() => {});
  };
};

export const getTermsandcondition = () => {
  return async dispatch => {
    const path = '/termsandconditions';
    openApiInterceptorForGet(
      dispatch,
      path,
      response => {
        dispatch({type: TERMS_AND_CONDITIONS, data: response.data});
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const updateTermsAndConditions = (input, successCallback) => {
  return async dispatch => {
    const path = '/termsandconditions';
    apiInterceptorForPost(
      dispatch,
      path,
      input,
      response => {
        successCallback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const deleteUser = (userRole, successCallback) => {
  return async dispatch => {
    const path = `/deleteuser/${userRole.type}`;
    apiInterceptorForPost(
      dispatch,
      path,
      {},
      response => {
        successCallback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const getLatestVersion = (successCallback?: any) => {
  return async dispatch => {
    const path = '/latestappversion';
    openApiInterceptorForGet(
      dispatch,
      path,
      response => {
        dispatch({type: LATEST_VERSION, data: response.data});
        successCallback && successCallback(response.data);
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const setStepTitles = stepTitles => {
  return async dispatch => {
    dispatch({type: SET_STEP_TITLES, data: stepTitles});
  };
};

export const setUserRole = userRole => {
  return async dispatch => {
    dispatch({type: SET_USER_ROLE, data: userRole});
  };
};

export const setShowRole = showRole => {
  return async dispatch => {
    dispatch({type: SET_SHOW_ROLE, data: showRole});
  };
};

export const setUserName = userName => {
  return async dispatch => {
    dispatch({type: SET_USER_NAME, data: userName});
  };
};

export const setUserAddress = userAddress => {
  return async dispatch => {
    dispatch({type: SET_USER_ADDRESS, data: userAddress});
  };
};

export const resetUser = () => {
  return async dispatch => {
    dispatch({type: RESET_USER});
  };
};