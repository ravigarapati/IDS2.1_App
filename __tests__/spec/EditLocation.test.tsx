import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import EditLocation from '../../src/pages/EditLocation';
import {render, cleanup, fireEvent} from 'react-native-testing-library';
import HomeOwnerReducer from './../../src/store/reducers/HomeOwnerReducer';
import * as homeOwnerActions from './../../src/store/actions/HomeOwnerActions';
import {createStore, combineReducers, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';

jest.mock('../../src/components/Button', () => 'Button');
jest.mock('../../src/components/CustomInputText', () => 'CustomInputText');
jest.mock('../../src/components/CustomText', () => 'CustomText');

const rootReducer = combineReducers({
  homeOwner: HomeOwnerReducer,
});

const realStore = createStore(
  rootReducer,
  {
    homeOwner: {
      energyData: {},
      deviceList: [],
      selectedDevice: {
        lockDevice: false,
        d_hour: '1',
        isFahrenheit: true,
        deadband: 3,
        schedules: [
          {
            default_home: '1',
            limit: '71-81',
            mode: '1',
            model_id: '1',
            name: 'Home',
            state: '0',
          },
          {
            default_home: '1',
            limit: '71-81',
            mode: '1',
            model_id: '2',
            name: 'Vacation',
            state: '0',
          },
          {
            default_home: 0,
            limit: '71-82',
            mode: '3',
            model_id: '3',
            name: 'summer ',
            state: '1',
          },
          {
            default_home: 0,
            limit: '71-82',
            mode: '3',
            model_id: '4',
            name: 'summer cam',
            state: '0',
          },
        ],
      },
      contractorMonitoringStatus: false,
      fanMode: '',
      fanOnFor: '',
      fanOffFor: '',
      fanIsScheduled: '',
      fanScheduledStart: '',
      fanScheduledEnd: '',
      selectedSchedule: '1',
      selectedScheduleName: '',
      scheduleInfo: {
        items1: [{c: '0', h: '0', t: '78.0-70.0'}],
        items2: [
          {c: '0', h: '0', t: '78.0-73.0'},
          {c: '0', h: '0', t: '99.0-90.0'},
        ],
        items3: [{c: '0', h: '1185', t: '52.0-48.0'}],
        items4: [{c: '0', h: '0', t: '78.0-70.0'}],
        items5: [{c: '0', h: '0', t: '78.0-70.0'}],
        items6: [{c: '0', h: '0', t: '78.0-70.0'}],
        items7: [{c: '0', h: '0', t: '78.0-70.0'}],
      },
      prevSelectedDevice: {},
      deviceStatus: {},
      deviceInformation: {},
      deviceDetails: {},
      heatPumpInfo: {},
      reloadHeatPumpInfo: false,
      notificationList: [],
      lastKey: {},
      notificationsCount: 0,
      lastUpdated: null,
      isTermsConditionsAccepted: null,
      hoUserAnalytics: null,
      selectedUnitName: {},
      isThermostatSelected: true,
      isUserFirstLogin: true,
      actualWeatherOnFahrenheit: true,
      hapticVibration: true,
      _24HrsFormatSelected: false,
      location: 'Alabaster',
      updatedSelectedDevice: false,
      weatherInfo: {},
      weatherInfoLocation: {},
      locationSuggestions: [
        {
          PlaceId:
            'AQABAFMAKZmJqReTuJvQ0SYSvNXBFPnTNre8NpZN_vO5ppVBiIVKMWyFo7U_23WZIEB3jBaCX9aLHNHy6gHy6QyfVv1o8QfoXctScRaEWzssnUy2A7vSsswLy5wkja6Jfp6hQXlv4QvGMRLRNcLPAogsltpxpywjew',
          Text: 'New York, NY, United States',
        },
      ],
      locationInformation: {},
      idsSelectedDeviceAccess: '',
      utilyEnergySaving: '',
      idsSelectedDevice: '',
      idsSelectedOdu: '',
      idsSelectedDeviceType: '',
      tempLocation: '',
      newDeviceInfo: {},
      previousBcc: '',
      deviceList2: [
        /*{
        id: 2,
        //macId: 'e8fdf8a4773c',
        macId: '34eae7c351e6',
        deviceName: 'Bedroom Thermostat',
        deviceType: 'BCC100',
        isThermostat: true,
        isOn: true,
        mode: 0,
        setPoint: 74,
        heatingSetpoint: 74,
        coolingSetpoint: 79,
        current: 76,
        isAccesories: true,
        accesories: [
          {
            id: 0,
            name: 'Dehumidifier',
            isEnable: 0,
            isOn: false,
          },
          {
            id: 1,
            name: 'Humidifier',
            isEnable: 1,
            isOn: false,
          },
        ],
        isMonitoring: null,
        isOnSchedule: true,
        stage: 3,
        isConnected: true,
        acceleratedHeating: true,
        schedules: [],
      },*/
      ],
    },
  },
  applyMiddleware(thunk),
);

const mockStore = configureMockStore();
const store = mockStore({
  homeOwner: {
    energyData: {},
    deviceList: [],
    selectedDevice: {},
    contractorMonitoringStatus: false,
    fanMode: '',
    fanOnFor: '',
    fanOffFor: '',
    fanIsScheduled: '',
    fanScheduledStart: '',
    fanScheduledEnd: '',
    selectedSchedule: '0',
    selectedScheduleName: '',
    scheduleInfo: {},
    prevSelectedDevice: {},
    deviceStatus: {},
    deviceInformation: {},
    deviceDetails: {},
    heatPumpInfo: {},
    reloadHeatPumpInfo: false,
    notificationList: [],
    lastKey: {},
    notificationsCount: 0,
    lastUpdated: null,
    isTermsConditionsAccepted: null,
    hoUserAnalytics: null,
    selectedUnitName: {},
    isThermostatSelected: true,
    isUserFirstLogin: true,
    actualWeatherOnFahrenheit: true,
    hapticVibration: true,
    _24HrsFormatSelected: false,
    location: 'Alabaster',
    updatedSelectedDevice: false,
    weatherInfo: {},
    weatherInfoLocation: {},
    locationSuggestions: [
      {
        PlaceId:
          'AQABAFMAKZmJqReTuJvQ0SYSvNXBFPnTNre8NpZN_vO5ppVBiIVKMWyFo7U_23WZIEB3jBaCX9aLHNHy6gHy6QyfVv1o8QfoXctScRaEWzssnUy2A7vSsswLy5wkja6Jfp6hQXlv4QvGMRLRNcLPAogsltpxpywjew',
        Text: 'New York, NY, United States',
      },
      {
        Text: 'New York Fries',
      },
    ],
    locationInformation: {},
    idsSelectedDeviceAccess: '',
    utilyEnergySaving: '',
    idsSelectedDevice: '',
    idsSelectedOdu: '',
    idsSelectedDeviceType: '',
    tempLocation: '',
    newDeviceInfo: {},
    previousBcc: '',
    deviceList2: [
      /*{
        id: 2,
        //macId: 'e8fdf8a4773c',
        macId: '34eae7c351e6',
        deviceName: 'Bedroom Thermostat',
        deviceType: 'BCC100',
        isThermostat: true,
        isOn: true,
        mode: 0,
        setPoint: 74,
        heatingSetpoint: 74,
        coolingSetpoint: 79,
        current: 76,
        isAccesories: true,
        accesories: [
          {
            id: 0,
            name: 'Dehumidifier',
            isEnable: 0,
            isOn: false,
          },
          {
            id: 1,
            name: 'Humidifier',
            isEnable: 1,
            isOn: false,
          },
        ],
        isMonitoring: null,
        isOnSchedule: true,
        stage: 3,
        isConnected: true,
        acceleratedHeating: true,
        schedules: [],
      },*/
    ],
  },
  auth: {},
});

describe('Edit Location screen', () => {
  test('Component to be defined', () => {
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <EditLocation navigation={navigation} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <EditLocation navigation={navigation} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Click on cancel button', () => {
    let currentScreen = '';
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      navigate: screen => (currentScreen = screen),
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = render(
      <Provider store={store}>
        <EditLocation navigation={navigation} />
      </Provider>,
    );
    let cancelButton = tree.getByTestId('cancel');
    fireEvent(cancelButton, 'press');

    expect(currentScreen).toBe('HomeTabs');
  });

  test('Input new location', () => {
    let currentScreen = '';
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      navigate: screen => (currentScreen = screen),
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = render(
      <Provider store={realStore}>
        <EditLocation navigation={navigation} />
      </Provider>,
    );
    let location = tree.getByTestId('location');
    fireEvent(location, 'change', 'New York');
    jest
      .spyOn(homeOwnerActions, 'getLocationSuggestions')
      .mockReturnValueOnce({});
    jest.spyOn(homeOwnerActions, 'updateLocation').mockReturnValueOnce({
      deviceAddress: {
        city: 'Sagaponack',
        country: 'United States',
        state: 'New York',
        zipcode: '11962',
      },
      message: 'Operation succeed',
      weatherLocation: {latitude: 40.94298, longitude: -72.27131},
    });

    let suggestion = tree.getByTestId('suggestion');
    fireEvent(suggestion, 'press');

    let submit = tree.getByTestId('submit');
    fireEvent(submit, 'press');
    expect(location.props.value).toBe('New York');
  });

  test('Input new location (onPressIn)', () => {
    let currentScreen = '';
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      navigate: screen => (currentScreen = screen),
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = render(
      <Provider store={realStore}>
        <EditLocation navigation={navigation} />
      </Provider>,
    );
    let location = tree.getByTestId('location');
    fireEvent(location, 'change', 'New York');
    jest
      .spyOn(homeOwnerActions, 'getLocationSuggestions')
      .mockReturnValueOnce({});
    jest.spyOn(homeOwnerActions, 'updateLocation').mockReturnValueOnce({
      deviceAddress: {
        city: 'Sagaponack',
        country: 'United States',
        state: 'New York',
        zipcode: '11962',
      },
      message: 'Operation succeed',
      weatherLocation: {latitude: 40.94298, longitude: -72.27131},
    });

    let suggestion = tree.getByTestId('suggestion');
    fireEvent(suggestion, 'press');

    let submit = tree.getByTestId('submit');
    fireEvent(submit, 'press');
    expect(location.props.value).toBe('New York');
  });

  test('Input new location (onPressIn)', () => {
    let currentScreen = '';
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      navigate: screen => (currentScreen = screen),
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = render(
      <Provider store={realStore}>
        <EditLocation navigation={navigation} />
      </Provider>,
    );
    let location = tree.getByTestId('location');
    fireEvent(location, 'change', 'New York');
    jest
      .spyOn(homeOwnerActions, 'getLocationSuggestions')
      .mockReturnValueOnce({});
    jest.spyOn(homeOwnerActions, 'updateLocation').mockReturnValueOnce({
      deviceAddress: {
        city: 'Sagaponack',
        country: 'United States',
        state: 'New York',
        zipcode: '11962',
      },
      message: 'Operation succeed',
      weatherLocation: {latitude: 40.94298, longitude: -72.27131},
    });

    let suggestion = tree.getByTestId('suggestion');
    fireEvent(suggestion, 'onPressIn');

    let submit = tree.getByTestId('submit');
    fireEvent(submit, 'press');
  });

  test('Input new location (onPressOut)', () => {
    let currentScreen = '';
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      navigate: screen => (currentScreen = screen),
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = render(
      <Provider store={realStore}>
        <EditLocation navigation={navigation} />
      </Provider>,
    );
    let location = tree.getByTestId('location');
    fireEvent(location, 'change', 'New York');
    jest
      .spyOn(homeOwnerActions, 'getLocationSuggestions')
      .mockReturnValueOnce({});
    jest.spyOn(homeOwnerActions, 'updateLocation').mockReturnValueOnce({
      deviceAddress: {
        city: 'Sagaponack',
        country: 'United States',
        state: 'New York',
        zipcode: '11962',
      },
      message: 'Operation succeed',
      weatherLocation: {latitude: 40.94298, longitude: -72.27131},
    });

    let suggestion = tree.getByTestId('suggestion');
    fireEvent(suggestion, 'onPressOut');

    let submit = tree.getByTestId('submit');
    fireEvent(submit, 'press');
    expect(location.props.value).toBe('New York');
  });
});
