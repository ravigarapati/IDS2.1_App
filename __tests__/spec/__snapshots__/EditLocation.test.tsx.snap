// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Location screen Component to be rendered 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/header_ribbon.png",
      }
    }
    style={
      Object {
        "height": 8,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <CustomInputText
          disableCache={false}
          onChange={[Function]}
          testID="location"
        />
      </View>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/search.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "marginLeft": "-6%",
          }
        }
      />
    </View>
    <View>
      <Button
        disabled={true}
        onPress={[Function]}
        testID="submit"
        text="Save"
        type="primary"
      />
      <Button
        onPress={[Function]}
        testID="cancel"
        text="Cancel"
        type="secondary"
      />
    </View>
  </View>
</View>
`;
